src/components/user/PersonalProfileEditForm.js

```diff
@@ -147,7 +163,7 @@ export default function PersonalProfileEditForm() {
                 <TableCell>
                   <TextField
                     name="name"
-                    value={formData.name}
+                    value={formData.name || userData.username}
```

Comment:
Making this change here will lead to some issues; because now we are setting the value of the text field; but back in our `handleProfileSubmit` function were are setting errors if `formData` is empty:

```
      setErrors({
        name: !formData.name ? 'Name is required' : '',
        email: !formData.email ? 'Email is required' : '',
      });
```

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/6b7c4791-e152-443c-8c74-8fc2a829d16c)

Your first thought might be to add a check in `setErrors` that also does the same `!formData.name || !userData.name` but I feel that starts to get complicated if we need to be updating multiple spots to do this kind of conditional check.

Instead, I would rather set the `formData` when we fist render this component to the values that we want it to be.

Setting values when the component first renders, can be done with a `useEffect` hook:

```javascript
  useEffect(() => {
    setFormData({
      name: userData?.username,
      email: userData?.email,
      oldPassword: '',
      newPassword: '',
      profilePicture: null,
    });
  }, [])
```

Reply:
I have added useEffect , I added as well userData as a dependency to rerun the effect when userData changes,I am not sure if I did correctly🫣